client/my-sites/plugins/plugins-discovery-page/index.jsx

```diff
@@ -24,63 +24,88 @@ import usePlugins from '../use-plugins';
  */
 const SHORT_LIST_LENGTH = 6;
 
-const UpgradeNudge = ( { selectedSite, sitePlan, isVip, jetpackNonAtomic, siteSlug } ) => {
+const UpgradeNudge = ( {
+	selectedSite,
+	sitePlan,
+	isVip,
+	jetpackNonAtomic,
+	siteSlug,
+	paidPlugins,
+} ) => {
+	const hasInstallPurchasedPlugins = useSelector( ( state ) =>
+		siteHasFeature( state, selectedSite?.ID, WPCOM_FEATURES_INSTALL_PURCHASED_PLUGINS )
+	);
+	const requiredPlansPurchasedPlugins = useSelector( ( state ) =>
+		getPlansForFeature( state, selectedSite?.ID, WPCOM_FEATURES_INSTALL_PURCHASED_PLUGINS )
+	);
+	const requiredPlansAllPlugins = useSelector( ( state ) =>
+		getPlansForFeature( state, selectedSite?.ID, FEATURE_INSTALL_PLUGINS )
+	);
+
 	const translate = useTranslate();
+
 	const eligibleForProPlan = useSelector( ( state ) =>
 		isEligibleForProPlan( state, selectedSite?.ID )
 	);
 
-	if ( ! selectedSite?.ID || ! sitePlan || isVip || jetpackNonAtomic ) {
-		return null;
-	}
-	const isLegacyPlan = isBlogger( sitePlan ) || isPersonal( sitePlan ) || isPremium( sitePlan );
-	const checkoutPlan = eligibleForProPlan && ! isLegacyPlan ? 'pro' : 'business';
-	const bannerURL = `/checkout/${ siteSlug }/${ checkoutPlan }`;
-	const plan = findFirstSimilarPlanKey( sitePlan.product_slug, {
-		type: TYPE_BUSINESS,
-	} );
+	const requiredPlans = useSelector( ( state ) =>
+		getPlansForFeature( state, selectedSite?.ID, WPCOM_FEATURES_INSTALL_PURCHASED_PLUGINS )
+	);
 
-	const title =
-		eligibleForProPlan && ! isLegacyPlan
-			? translate( 'Upgrade to the Pro plan to install plugins.' )
-			: translate( 'Upgrade to the Business plan to install plugins.' );
+	// Whether to show an upgrade banner specific to premium plugins.
+	const lowerPlanAvailable =
+		! hasInstallPurchasedPlugins &&
+		requiredPlansPurchasedPlugins[ 0 ] !== requiredPlansAllPlugins[ 0 ];
 
-	return (
-		<UpsellNudge
-			event="calypso_plugins_browser_upgrade_nudge"
-			showIcon={ true }
-			href={ bannerURL }
-			feature={ FEATURE_INSTALL_PLUGINS }
-			plan={ plan }
-			title={ title }
-		/>
-	);
-};
+	let bannerURL;
+	let feature;
+	let plan;
+	let title;
 
-const UpgradeNudgePaid = ( props ) => {
-	const translate = useTranslate();
+	if ( paidPlugins && ! hasInstallPurchasedPlugins && lowerPlanAvailable ) {
+		if ( ! requiredPlans ) {
+			return null;
+		}
 
-	const requiredPlans = useSelector( ( state ) =>
-		getPlansForFeature( state, props.selectedSite?.ID, WPCOM_FEATURES_INSTALL_PURCHASED_PLUGINS )
-	);
+		const requiredPlan = getPlan( requiredPlans[ 0 ] );
 
-	if ( ! requiredPlans ) {
+		bannerURL = `/checkout/${ siteSlug }/${ requiredPlan.getPathSlug() }`;
+		feature = WPCOM_FEATURES_INSTALL_PURCHASED_PLUGINS;
+		plan = requiredPlan.getStoreSlug();
+		title = translate( 'Upgrade to the %(planName)s plan to install premium plugins.', {
+			textOnly: true,
+			args: { planName: requiredPlan.getTitle() },
+		} );
```

Comment:
If the props are just being set once it's usually better to return the component instead of deferring. e.g return a `<UpsellNudge ... />` here instead.

Reply:
✅